src/site/content/en/blog/kapwing/index.md

```diff
@@ -0,0 +1,415 @@
+---
+
+Online video consumption has grown rapidly since the start of the pandemic.
+People are spending more time consuming endless high-quality video on
+platforms such as TikTok, Instagram, and YouTube. Creatives and small business
+owners all over the world need quick and easy-to-use tools to make video
+content. 
+
+Companies like Kapwing make it possible to create all this video content right
+on the web, by using the latest in powerful APIs and performance tools. 
+
+## About Kapwing
+
+Kapwing is a web-based collaborative video editor designed mainly for casual
+creatives like game-streamers, musicians, YouTube creators, and meme-rs. It's
+also a go-to resource for business owners who need an easy way to produce
+their own social content, such as Facebook and Instagram ads.
+
+People discover Kapwing by searching for a specific task, for example "how to
+trim a video," "add music to my video," or "resize a video." They can do what
+they searched for with just one click&mdash;without the added friction of
+navigating to an app store and downloading an app. The web makes it simple for
+people to search for precisely what task they need help with, and then do it.
+
+After that first click, Kapwing users can do a whole lot more just as easily.
+They can explore free templates; add new layers of free stock videos, insert
+subtitles, transcribe videos, and upload background music. 
+
+## How Kapwing brings real-time editing and collaboration to the web
+
+While the web provides unique advantages, it also presents distinct
+challenges. Kapwing needs to deliver smooth and precise playback of complex,
+multi-layered projects across a wide range of devices and network conditions.
+
+To this end we use a variety of web APIs to achieve our performance and
+feature goals.
+
+### IndexedDB
+
+High performance editing requires that all of our users' content live on the
+client, avoiding the network whenever possible. Unlike a streaming service,
+where users typically access a piece of content once, our customers reuse
+their assets frequently, days and even months after upload.
+
+IndexedDB allows us to provide persistent file system-like storage to our
+users. The result is that over 90% of media requests in the app are fulfilled
+locally. Integrating indexedDB into our system was very straightforward.
+
+Here is some boiler plate initialization code that runs on app load:
+
+```js
+import { DBSchema, openDB, deleteDB, IDBPDatabase } from "idb";
+let openIdb: Promise<IDBPDatabase<Schema>>;
+const db =
+  (await openDB) <
+  Schema >
+  ("kapwing",
+  version,
+  {
+    upgrade(db, oldVersion) {
+      if (oldVersion >= 1) {
+        // assets store schema changed, need to recreate
+        db.deleteObjectStore("assets");
+      }   
+      db.createObjectStore("assets", { keyPath: "mediaLibraryID" }); 
+    },  
+    async blocked() {
+      await deleteDB("kapwing");
+    },  
+    async blocking() {
+      await deleteDB("kapwing");
+  },
+});
+```
```

Comment:
```suggestion
```js
import {DBSchema, openDB, deleteDB, IDBPDatabase} from 'idb';

let openIdb: Promise <IDBPDatabase<Schema>>;

const db =
  (await openDB) <
  Schema >
  (
    'kapwing',
    version, {
      upgrade(db, oldVersion) {
        if (oldVersion >= 1) {
          // assets store schema changed, need to recreate
          db.deleteObjectStore('assets');
        }

        db.createObjectStore('assets', {
          keyPath: 'mediaLibraryID'
        });
      },
      async blocked() {
        await deleteDB('kapwing');
      },
      async blocking() {
        await deleteDB('kapwing');
      },
    }
  );
```
```

Reply:
These changes are to adhere with [Google's JavaScript style guide](https://google.github.io/styleguide/jsguide.html).

Reply:
I'm OOO so I don't want to look too closely: if the edits are only changing the formatting to make the code more readable, great! Thank you for the extra attention on this!

If you've suggested any changes that are prescriptive (like use `let` instead of `var`), someone should get Josh from Kapwing to sign off--it is their app and their code after all. @heyawhite @rowan-m to handle the comms around that.